@feature/controllers routes (#42)
bedeve/fullstack-development: packages/documentation/web-app/index.mdx

#### validations

#### controllers

> Controllers are functions called by routes. They get passed a request, perform the business logic, execute queries and mutations on the database and return a response.

Reviewer: ```suggestion
*remobe*<Controllers are functions called by routes. They get passed a request, perform the business logic, execute queries and mutations on the database and return a response.>
```

#### api routes

A route is an endpoint that relates an HTTP method (GET, POST, PUT, DELETE, etc.) on a URL path with the corresponding controller.

##### HTTP request methods

HTTP defines a set of *** request methods *** to indicate the desired action to be performed for a given resource. Although they can also be nouns, these request methods are sometimes referred as HTTP verbs. Each of them implements a different semantic, but some common features are shared by a group of them: e.g. a request method can be safe, idempotent, or cacheable.

*     GET

 The ```GET``` method requests a representation of the specified resource. Requests using GET should only retrieve data.

*     POST

    The ```POST``` method is used to submit (create) an entity to the specified resource, often causing a change in state or side effects on the server.

*     PUT

    The ```PUT``` method replaces (update) all current representations of the target resource with the request payload.

*     DELETE

    The ```DELETE``` method deletes the specified resource.

[More...](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)

> ##### URI (Uniform Resource Identifier)

Reviewer 2: ```suggestion
```

##### Create a new route and relate it to a controller (using the express.js framework)

To create a new route we call the *route()* method, defining the path as in the following example:

    router.route(`/api/songs`)

> inside the method, we use a request method (e.g. a **get** request) in which which then calls the main function of the controller

Reviewer 3: ```suggestion
inside the method, we use a request method (e.g. a **get** request) which calls the main function of the controller
```

#### api routes

A route is an endpoint that relates an HTTP method (GET, POST, PUT, DELETE, etc.) on a URL path with the corresponding controller.

##### HTTP request methods

HTTP defines a set of *** request methods *** to indicate the desired action to be performed for a given resource. Although they can also be nouns, these request methods are sometimes referred as HTTP verbs. Each of them implements a different semantic, but some common features are shared by a group of them: e.g. a request method can be safe, idempotent, or cacheable.

*     GET

 The ```GET``` method requests a representation of the specified resource. Requests using GET should only retrieve data.

*     POST

    The ```POST``` method is used to submit (create) an entity to the specified resource, often causing a change in state or side effects on the server.

*     PUT

    The ```PUT``` method replaces (update) all current representations of the target resource with the request payload.

*     DELETE

    The ```DELETE``` method deletes the specified resource.

[More...](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)

##### URI (Uniform Resource Identifier)

> A URI (Uniform Resource Identifier) is a string that refers to a resource and follows a predefined set of syntax rules. The most common are URLs, which identify the resource by giving its location on the Web.

Reviewer 4: ```suggestion

```

#### api routes

A route is an endpoint that relates an HTTP method (GET, POST, PUT, DELETE, etc.) on a URL path with the corresponding controller.

##### HTTP request methods

HTTP defines a set of *** request methods *** to indicate the desired action to be performed for a given resource. Although they can also be nouns, these request methods are sometimes referred as HTTP verbs. Each of them implements a different semantic, but some common features are shared by a group of them: e.g. a request method can be safe, idempotent, or cacheable.

*     GET

 The ```GET``` method requests a representation of the specified resource. Requests using GET should only retrieve data.

*     POST

    The ```POST``` method is used to submit (create) an entity to the specified resource, often causing a change in state or side effects on the server.

*     PUT

    The ```PUT``` method replaces (update) all current representations of the target resource with the request payload.

*     DELETE

    The ```DELETE``` method deletes the specified resource.

[More...](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)

##### URI (Uniform Resource Identifier)

> A URI (Uniform Resource Identifier) is a string that refers to a resource. The most common are URLs, which identify the resource by giving its location on the Web.

Reviewer 2: ```suggestion
A URI (Uniform Resource Identifier) is a string that refers to a resource and follows a predefined set of syntax rules. The most common are URLs, which identify the resource by giving its location on the Web.
```

> 

Reviewer 4: ```suggestion

```

> 

Reviewer 4: ```suggestion
```

##### Create a new route and relate it to a controller (using the express.js framework)

> To create a new route we call the *route* method, defining the path as in the following example:

Reviewer 3: ```suggestion
To create a new route we call the *route()* method, defining the path as in the following example:
```

#### validations

#### controllers

> Controllers are functions used to get the requested data from the models, create an HTML page displaying the data, and return it to the user to view in the browser.

Reviewer 5: ```suggestion
Controllers are functions called by routes. They get passed a request, perform the business logic, execute queries and return a response.
```

Reviewer 4: ```suggestion
Controllers are functions called by routes. They get passed a request, perform the business logic, execute queries and mutations on the database and return a response.
```

#### api routes

> A route is an endpoint that connects a method (GET, POST, PUT, DELETE, etc.), a URL  path/pattern and the controller.

Reviewer 5: ```suggestion
A route is an endpoint that connects a method (GET, POST, PUT, DELETE, etc.), a URL  path/pattern and the controller.
```
```suggestion
A route is an endpoint that connects a method (GET, POST, PUT, DELETE, etc.), a URL  path/pattern and the controller.
```
```suggestion
A route is an endpoint that relates an HTTP method (GET, POST, PUT, DELETE, etc.) on a URL path with the corresponding controller.
```

> A route is a section of Express code that associates an HTTP verb (GET, POST, PUT, DELETE, etc.), an URL path/pattern, and a function that is called to handle that pattern.

Reviewer 3: ```suggestion
A route is an endpoint that connects a method (GET, POST, PUT, DELETE, etc.), a URL  path/pattern and the controller.
```

Reviewer 5: ```suggestion
A route is an endpoint that relates an HTTP method (GET, POST, PUT, DELETE, etc.) on a URL path with the corresponding controller.
```

##### HTTP request methods

HTTP defines a set of *** request methods *** to indicate the desired action to be performed for a given resource. Although they can also be nouns, these request methods are sometimes referred as HTTP verbs. Each of them implements a different semantic, but some common features are shared by a group of them: e.g. a request method can be safe, idempotent, or cacheable.

*     GET

 The ```GET``` method requests a representation of the specified resource. Requests using GET should only retrieve data.

*     POST

>     The ```POST``` method is used to submit an entity to the specified resource, often causing a change in state or side effects on the server.

Reviewer 2: ```suggestion
    The ```POST``` method is used to submit (create) an entity to the specified resource, often causing a change in state or side effects on the server.
```

*     PUT

>     The ```PUT``` method replaces all current representations of the target resource with the request payload.

Reviewer 2: ```suggestion
    The ```PUT``` method replaces (update) all current representations of the target resource with the request payload.
```

*     DELETE

    The ```DELETE``` method deletes the specified resource.

[More...](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)

##### URI (Uniform Resource Identifier)

A URI (Uniform Resource Identifier) is a string that refers to a resource. The most common are URLs, which identify the resource by giving its location on the Web.

##### Create a new route and relate it to a controller (using the express.js framework)

To create a new route we call the *route* method, defining the path as in the following example:

    router.route(`/api/songs`)

inside the method, we use a request method (e.g. a **get** request) in which which then calls the main function of the controller

example: 

    .get(
         search

>         

Reviewer 5: ```suggestion
```
